test/modules/post/test/file.rb

```diff
@@ -46,6 +47,76 @@ def setup
     super
   end
 
+  def test_dir
+    fs_sep = session.platform == 'windows' ? '\\' : '/'
+
+    it 'should test for directory existence' do
+      ret = false
+      [
+        'c:\\',
+        '/etc/',
+        '/tmp'
+      ].each do |path|
+        ret = true if directory?(path)
+      end
+
+      ret
+    end
+
+    it 'should create directories' do
+      mkdir(datastore['BaseDirectoryName'])
+      ret = directory?(datastore['BaseDirectoryName'])
+      ret &&= write_file([datastore['BaseDirectoryName'], 'file'].join(fs_sep), '')
+      ret &&= mkdir([datastore['BaseDirectoryName'], 'directory'].join(fs_sep))
+      ret &&= write_file([datastore['BaseDirectoryName'], 'directory', 'file'].join(fs_sep), '')
+      ret
+    end
+
+    it 'should list the directory we just made' do
+      dents = dir(datastore['BaseDirectoryName'])
+      dents.include?('file') && dents.include?('directory')
+    end
+
+    it 'should recursively delete the directory we just made' do
+      rm_rf(datastore['BaseDirectoryName'])
+      !directory?(datastore['BaseDirectoryName'])
+    end
+
+    if session.platform != 'windows' && command_exists?('ln')
```

Comment:
Do we want to throw on `mklink` tests for windows too? 👀 

Just for a potential future CSharp meterpreter, it looks like [Directory.CreateDirectory](https://docs.microsoft.com/en-us/dotnet/api/system.io.directory.createdirectory?view=net-6.0) creates paths recursively by default, whilst [Directory.Delete](https://docs.microsoft.com/en-us/dotnet/api/system.io.directory.delete?view=net-6.0#system-io-directory-delete(system-string-system-boolean) requires an extra boolean flag to delete recursively - and there's some notes on how symlinks are handled in the remarks section too. Potentially a useful future proof test if OJ's [clr-meterpreter](https://github.com/OJ/clr-meterpreter/blob/c806cc807e81f794cabab386278cf737bc2de25b/src/stdapi.net35/FileSystem.cs#L210) is revived, as it doesn't specify that flag from what I can see

Reply:
I looked into how command shells and the Windows Meterpreter would handle deleting a symbolic link and neither of them will delete a symbolic link target. If we define this behavior by adding the test, the Windows Meterpreter is going to need to be updated for this along with the [Windows Command](https://github.com/rapid7/metasploit-framework/blob/c154130832771c1bdf0e68ecb142e728ac1f9b98/lib/msf/core/post/file.rb#L576) handler, possibly the Powershell one too. If we could, I'd prefer to track symbolic link behavior on the Windows platform as a task separate from how non-empty directories are handled on all platforms.

Reply:
Added in 6b64985328ba442b2212357c8132a2cd21cc60bb. The Windows Meterpreter passes, Windows Shell sessions pass, the Python Meterpreter throws an error when running `cmd.exe /c mklink` and Powershell sessions are failing as well.